src/Modules/OrchardCore.Commerce/Drivers/OrderPartDisplayDriver.cs

```diff
@@ -55,10 +78,30 @@ private async ValueTask BuildViewModelAsync(OrderPartViewModel model, OrderPart
                 UnitPrice = lineItem.UnitPrice,
                 LinePrice = lineItem.LinePrice,
                 ProductRouteValues = metaData.DisplayRouteValues,
-                Attributes = lineItem.Attributes.ToDictionary(attr => attr.Key, attr => attr.Value),
+                Attributes = lineItem.Attributes,
             };
         }));
-        foreach (var item in lineItems) model.LineItems.Add(item);
+
+        var total = new Amount();
+
+        foreach (var item in lineItems)
+        {
+            model.LineItems.Add(item);
+
+            if (total.Currency.Equals(Currency.UnspecifiedCurrency))
+            {
+                total = item.LinePrice;
+            }
+            else
+            {
+                total += item.LinePrice;
+            }
```

Comment:
I don't understand the logic here. Why does a value with unspecified currency reset the total?

Reply:
It was for initializing the total. But it's stupid, I replaced it with this:
```
        var total = new Amount(0, lineItems.FirstOrDefault().LinePrice.Currency);

        foreach (var item in lineItems)
        {
            model.LineItems.Add(item);

            total += item.LinePrice;
        }
```